imagetest/test_suites/disk/setup.go

```diff
@@ -30,5 +47,21 @@ func TestSetup(t *imagetest.TestWorkflow) error {
 		}
 	}
 	vm.RunTests("TestDiskReadWrite|TestDiskResize")
+	// Block device naming is an interaction between OS and hardware alone on windows, there is no guest-environment equivalent of udev rules for us to test.
+	if !utils.HasFeature(t.Image, "WINDOWS") {
+		for _, tc := range blockdevNamingCases {
+			if tc.arch != t.Image.Architecture {
+				continue
+			}
+			inst := &daisy.Instance{}
+			inst.MachineType = tc.machineType
+			inst.Name = "test-block-naming-" + tc.machineType
+			vm, err := t.CreateTestVMMultipleDisks([]*compute.Disk{{Name: inst.Name, Type: imagetest.PdBalanced}, {Name: "secondary", Type: imagetest.PdBalanced, SizeGb: 10}}, inst)
+			if err != nil {
+				return err
+			}
+			vm.RunTests("TestBlockDeviceNaming")
```

Comment:
Non related to this PR note: we could change this `RunTests()` to instead of receiving strings with the tests names receive function pointers so we can check if the test name is correct in build time rather than in run time.

Reply:
Ack. I agree with you that it would be good to move away from this. There are two barriers:
1) The test execution happens inside the VM after downloading the test binary, we can't pass a pointer acrossthis boundary. We need a (probably home grown) system for identifying and referring to test functions.
2) there's a lot of assumptions in CIT around the "build a `go test` binary and execute it inside the VM" architecture

I don't think there's any quick fixes for this, it would involve refactoring the test wrapper and modifying the result fetching/generation at a minimum.

Reply:
Agreed, thanks for clarifying.